channel/channel.go

```diff
@@ -2,93 +2,166 @@ package channel
-		Logger.Info().Msgf("%s %X", UnblockedPush, c.c)
+		c.log.Info().Msgf("unblocked channel %X on send", c.c)
 	}
 }
 
-// PushWithTimeout adds an element in the channel,
-// or logs a warning if it fails after the given timeout
-func (c *Timed[T]) PushWithTimeout(t time.Duration, e T) {
+// SendWithTimeout adds an element in the channel,
+// or logs a warning if it fails after the given timeout.
+// Note: this is a blocking call as it waits on a channel.
+func (c *Timed[T]) SendWithTimeout(t time.Duration, e T) {
 	ctx, cancel := context.WithTimeout(context.Background(), t)
 	defer cancel()
 
-	c.PushWithContext(ctx, e)
+	c.SendWithContext(ctx, e)
 }
 
-// Push adds an element in the channel,
-// or logs a warning if it fails after default timeout
-func (c *Timed[T]) Push(e T) {
+// Send adds an element in the channel,
+// or logs a warning if it fails after default timeout.
+// Note: this is a blocking call as it waits on a channel.
+func (c *Timed[T]) Send(e T) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultChannelTimeout)
 	defer cancel()
 
-	c.PushWithContext(ctx, e)
+	c.SendWithContext(ctx, e)
 }
 
-// PopWithContext removes an element from the channel
-// or logs a warning if it fails after the given context
-func (c *Timed[T]) PopWithContext(ctx context.Context) T {
+// ReceiveWithContext removes an element from the channel
+// or logs a warning if it fails in the given context.
+// Note: this is a blocking call as it waits on a channel.
+func (c *Timed[T]) ReceiveWithContext(ctx context.Context) T {
 	var e T
 
 	select {
 	case e = <-c.c:
 	case <-ctx.Done():
-		Logger.Warn().Msgf("%s %X\n%s", BlockedPop, c.c, string(debug.Stack()))
+		c.log.Warn().Msgf("%s %X\n%s", ErrFailedToReceive, c.c, string(debug.Stack()))
 		c.c <- e
-		Logger.Info().Msgf("%s %X", UnblockedPop, c.c)
+		c.log.Info().Msgf("unblocked channel %X on receiving", c.c)
 	}
 
 	return e
 }
 
-// PopWithTimeout removes an element from the channel
-// or logs a warning if it fails after the given timeout
-func (c *Timed[T]) PopWithTimeout(t time.Duration) T {
+// ReceiveWithTimeout removes an element from the channel
+// or logs a warning if it fails after the given timeout.
+// Note: this is a blocking call as it waits on a channel.
+func (c *Timed[T]) ReceiveWithTimeout(t time.Duration) T {
+	ctx, cancel := context.WithTimeout(context.Background(), t)
+	defer cancel()
+
+	return c.ReceiveWithContext(ctx)
+}
+
+// Receive removes an element from the channel
+// or logs a warning if it fails after the default timeout.
+// Note: this is a blocking call as it waits on a channel.
+func (c *Timed[T]) Receive() T {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultChannelTimeout)
+	defer cancel()
+
+	return c.ReceiveWithContext(ctx)
+}
+
+// NonBlockingSendWithContext adds an element in the channel,
+// or returns an error if it fails in the given context.
+func (c *Timed[T]) NonBlockingSendWithContext(ctx context.Context, e T) error {
```

Comment:
I'm not sure I understand. The name suggest this is a non-blocking operation, but it does exactly the same as `SendWithContext`, except it returns an error instead of logging it.